### ISSUE TITLE
fftn on 2D array with axes=[0] applies FFT to wrong axis

### ISSUE THREAD

**Issue by shoyer** [google/jax#1878]
I'm not sure how this snuck through our unit tests!

```python
import jax.numpy as jnp
import numpy as np

signal = np.ones((2, 2))
print('numpy result:\n', np.fft.fftn(signal, axes=[0]).real)
print('jax.numpy result:\n', jnp.fft.fftn(signal, axes=[0]).real)
```
Prints:
```
numpy result:
 [[2. 2.]
 [0. 0.]]
jax.numpy result:
 [[2. 0.]
 [2. 0.]]
```
